### DISCUSSION TITLE
Project based on FastAccelStepper

### DISCUSSION THREAD

**Discussion by kthod861** [gin66/FastAccelStepper#130] [Show and tell]
Hey,

You are probably interested about what people do with your awesome lib.

Here it is : https://youtu.be/-X5VyPOTz5s 

Cheers

**Comment by gin66** [comment]
Thanks for letting me know. Actually Youtube has highlighted this new video to me already before.
Looks really nice. Curious about your next video

**Comment by Ozark-iot** [comment]
Hi, we have a project using ESP32-S3-DevKitC-1 and ESP32-C3. These I understand are the newest offerings from Espressif.

However your FastAccelStepper library will not compile using the Arduino IDE if I select either of the above mentioned boards.

It would be fantastic if you would let me know if you plan to support these boards.

**Reply by gin66** [reply]
apparently S3 support is ready

**Reply by sp3swj** [reply]
I read that ESP32-C3 is not suported yet

https://github.com/gin66/FastAccelStepper#esp32c3esp32-mini-1

maybe some good news ??  about C3  ?  :-)

This module looks very nice - Do you have any plan to add this module to "supported" ?

https://shop.m5stack.com/products/m5stamp-c3u-mate-with-pin-headers

//EDIT

I found that can buy M5 with ESP32-S3 - and this solve "problem"
https://shop.m5stack.com/products/m5stamp-esp32s3-module

**Reply by gin66** [reply]
I have received the SuperMini ESP32C3 boards and hope to find the time till end of October for a working implementation.

**Reply by sp3swj** [reply]
:-) I bought C3 ... before read compatibility list  :-)  now in basket  S3 - wait for confirm order.

![image](https://github.com/gin66/FastAccelStepper/assets/4106517/a048dd81-c194-43b7-90f8-62d41edea53c)

**Reply by sp3swj** [reply]
M5 STAMP ESP32 S3  on desktop ;-)
200kHz works, 2motor works simultaneously on small code prepared manually. 

 But I can not load proper brogram wiglth "big demo", program compile, loading but cpu M5 Stamp ESP32 S3 not starting. This same code compiled for NANO works fine.

I will prepare video to show problem.

**Reply by gin66** [reply]
STM32 is no esp32 ?!

**Reply by gin66** [reply]
the esp32c3 version has been pushed to github. not yet officially released

**Reply by sp3swj** [reply]
Edit my error , naturally ESP32

**Comment by sp3swj** [comment]
Next projects found on YT

https://www.youtube.com/watch?v=fDHmkd_kX_E
https://www.youtube.com/watch?v=YsLykxnHApg
https://www.youtube.com/watch?v=7TkybpSQULk

:-)

**Reply by gin66** [reply]
Thanks for sharing. The last one is already listed in the readme. The first one, I am not sure. The second is about AccelStepper, right ?

**Reply by sp3swj** [reply]
ahhh  my fault - second is on different  - anyway maybe will be usefull for someone

Good news - few minutes ago I started "FAS"  :-) mean Your library on NANO - works, start learning how to use - how limitations.

A lot of time i waste for Windows 11 + Arduino + CH340 - but now solved - work in progress

**Reply by DaveAuld** [reply]
@gin66 FYI some of the links to those YT video are no longer available videos, so probably want to remove from the readme.

**Comment by DerekESchulte** [comment]
I'm thinking of trying to port to the Sparkfun SAMD51 Thing Plus (ATSAMD51J20 at 120MHz). This looks like it might be a straightforward adaptation of @clazarowitz 's Due port...but I may be underestimating the effort as the Due's AT91SAM3X8E may have more differences than I realize.  I'd also be willing to pay somebody to do it...Any advice?  [I'm using single DRV8711 or MP6500 driver depending on the hardware set.]

**Reply by sp3swj** [reply]
look at this small module  - https://shop.m5stack.com/products/m5stamp-esp32s3-module - maybe will fit you needs  :-)

can serve upt four motor  :-)  https://github.com/gin66/FastAccelStepper#esp32s3

**Reply by DerekESchulte** [reply]
Alas, too late to switch modules.  If I were to do that at this point I'd probably go to a Teensy3.5 and TeensyStep...

**Reply by clazarowitz** [reply]
The due port uses the pwm peripheral to generate the pulses for the stepper.  This seems to be pretty unique to the due.  The 51 doesn’t have one, so it’s not a straightforward port. 

Tbh the due port isn’t wonderful. I realize that now since I  have a better understanding of the datasheet and their overuse of duplicate acronyms in the same paragraph, so I’d like to redo it.  I’d also like to do a timer port, as the nxp (I believe the teensy uses nxp’s) and the stm microcontrollers all have many timers, but no pwm peripheral.  Ardupilot would like to have native stepper code in it, and a timer port would allow that on the pixhawks, which would obviate the need for the due in my project :)

I haven’t been in the code though because that project is a low priority project (others with the potential to make money take priority over fun….sometimes :) ) and because that project has suffered a terrible mechanical engineer (me).  First I tried using some limit switches, but Hall effect…which, well, on a steel lawn mower didn’t work as well as I had hoped) then a magnetic rotary encoder, which showed me the belts stretch far too much when loaded to be useful, so now I’m on a chain drive, and that seems to be working…finally!  So I’m onto calibrating straight line tracking this weekend…which will involve using the code, so I may get back into it :)

**Reply by DerekESchulte** [reply]
Thank you for the reply. I clearly didn't look at the code and datasheets close enough---I saw the use of the "advanced" timer TCCx, which both chips have and that can generate PWM and there's also a PIO vs. PORT nomenclature change for somewhat similar function. I'd be happy to pay you $1k and send a few CUI capacitive encoders (sweet and robust) and SAMD51-MP6500 your way to get it done, but that's also peanuts in comparison to real wages.
